packages/quick_start/source/SampleApp.ts

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import fs from "fs";
+import Utils, { Authentication, Ingestion, Queries } from "./Utils";
+import Console from "console";
+import KustoClient from "azure-kusto-data/source/client";
+import IngestClient, { KustoIngestClient } from "azure-kusto-ingest/source/ingestClient";
+import { DataFormat } from "azure-kusto-ingest";
+import readline from "readline";
+import { dataFormatMappingKind } from "azure-kusto-ingest/source/ingestionProperties";
+import { v4 as uuidv4 } from "uuid";
+
+enum SourceType {
+    LocalFileSource = "localfilesource",
+    BlobSource = "blobsource",
+}
+
+export enum AuthenticationModeOptions {
+    UserPrompt = "UserPrompt",
+    ManagedIdentity = "ManagedIdentity",
+    AppKey = "AppKey",
+    AppCertificate = "AppCertificate",
+}
+
+interface ConfigData {
+    sourceType: SourceType;
+    dataSourceUri: string;
+    format: string;
+    useExistingMapping: boolean;
+    mappingName: string;
+    mappingValue: string;
+}
+
+interface ConfigJson {
+    kustoUri: string;
+    ingestUri: string;
+    databaseName: string;
+    tableName: string;
+    useExistingTable: boolean;
+    alterTable: boolean;
+    queryData: boolean;
+    ingestData: boolean;
+    tableSchema: string;
+    data: ConfigData[];
+    certificatePath: string;
+    certificatePassword: string;
+    applicationId: string;
+    tenantId: string;
+    authenticationMode: AuthenticationModeOptions;
+    waitForUser: boolean;
+    waitForIngestSeconds: number;
+    batchingPolicy: string;
+}
+
+// TODO (config):
+//  If this quickstart app was downloaded from OneClick, kusto_sample_config.json should be pre-populated with your cluster's details.
+//  If this quickstart app was downloaded from GitHub, edit kusto_sample_config.json and modify the cluster URL and database fields appropriately.
+const CONFIG_FILE_NAME = "packages/quick_start/source/kusto_sample_config.json";
+
+class SampleApp {
+    private static step = 1;
+    private static waitForUser: boolean;
+
+    public static async start() {
+        Console.log("Kusto sample app is starting...");
+
+        const config = await this.loadConfigsAsync(CONFIG_FILE_NAME);
+        this.waitForUser = config.waitForUser;
+        if (config.authenticationMode === AuthenticationModeOptions.UserPrompt) {
+            await this.waitForUserToProceedAsync(
+                "You will be prompted *twice* for credentials during this script. Please return to the console after" + " authenticating."
+            );
+        }
+        const kustoConnectionString = await Authentication.generateConnectionStringAsync(
+            config.kustoUri,
+            config.authenticationMode,
+            config?.certificatePath,
```

Comment:
Why null check for the config here and not before?
I think they shouldn't be here at all